src/main/java/seedu/address/model/asset/Asset.java

```diff
@@ -0,0 +1,89 @@
+package seedu.address.model.asset;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.HashMap;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+/**
+ * Represents an asset in the address book.
+ * Guarantees: immutable.
+ */
+public class Asset {
+
+    /**
+     * Stores all unique assets created.
+     */
+    private static final HashMap<String, Asset> assetsHashMap = new HashMap<>();
+    private static final String MESSAGE_CONSTRAINTS = "Asset names should be alphanumeric";
+    private static final String VALIDATION_REGEX = "\\p{Alnum}+";
```

Comment:
Should we allow non-alphanumeric characters like `-` and `#`?

Reply:
Changed to any non empty name.